baseapp/genesis.go

```diff
@@ -0,0 +1,21 @@
+package baseapp
+
+import (
+	"github.com/cometbft/cometbft/abci/types"
+)
+
+// GenesisState allows modules to define a set of state transitions
+// that will initialize the chain's state at genesis.
+type GenesisState interface {
+	// SetState sets the genesis state.
+	// This should be called in a order define by the application developer
+	SetState([]byte) types.ResponseDeliverTx
```

Comment:
`SetState` is too generic of a name for this. The name should reflect that it is executing a transaction. Also, I don't think we need to return `ResponseDeliverTx`. All genutil needs to know is whether it was successful. How about:

```
type GenesisTxHandler interface {
  ExecuteGenesisTx([]byte) error
}
```

Ideally, we would have some sort of check in this function to make sure it can only execute during `InitChain` and not at any other time.

Reply:
thanks, ill make the changes. I think there is a way to do it, wanted to see if people are up for it before continuing